### ISSUE TITLE
Bidoo modules don't remember samples.

### ISSUE THREAD

**Issue by TheGuyWhoo** [DISTRHO/Cardinal#236]
I'm unfortunately unable to remember which build I'm using as I renamed the folder. But this appears to only affect bidoo modules. The sample is missing from the module whether it's a preset in my daw, a saved project in reaper, or a patch. It remembers slider/knob position, but the sample is no longer loaded when I save the patch through one of these methods and later come back to it after closing it. Sometimes it even causes a crash, like in this patch:

[crash_patch.zip](https://github.com/DISTRHO/Cardinal/files/8741132/crash_patch.zip)

Linux mint 20 64 bit, cinnamon, present in reaper and carla and standalone and all other variants(vst3, vst2, lv2)

**Comment by TheGuyWhoo** [comment]
Okay so I checked on the latest build from actions tab:

https://github.com/DISTRHO/Cardinal/commit/5eba049074e375c60674743e0004e7f14e233788

and that patch doesn't cause a crash anymore, though bidoo modules still don't remember the sample.

**Comment by offthesky** [comment]
i too am having the same issue with bidoo samplers losing the sample on reload. win10 ableton/reaper

**Comment by falkTX** [comment]
Seems like a bug in the bidoo modules, I opened your patch file and this is the section for limonade:

```
    {
      "id": 7608243557383411,
      "plugin": "Bidoo",
      "model": "liMonADe",
      "version": "2.0",
      "params": [
        {
          "value": 0.0,
          "id": 0
        },
        {
          "value": 1.0,
          "id": 1
        },
        {
          "value": -29.277133941650391,
          "id": 2
        },
        {
          "value": 0.0,
          "id": 3
        },
        {
          "value": 0.0,
          "id": 4
        },
        {
          "value": 0.0,
          "id": 5
        },
        {
          "value": 0.21530087292194366,
          "id": 6
        },
        {
          "value": 0.020266616716980934,
          "id": 7
        },
        {
          "value": 1.0,
          "id": 8
        },
        {
          "value": 0.010080003179609776,
          "id": 9
        },
        {
          "value": 0.0,
          "id": 10
        },
        {
          "value": 0.0,
          "id": 11
        },
        {
          "value": 0.0,
          "id": 12
        },
        {
          "value": 0.0,
          "id": 13
        },
        {
          "value": 0.0,
          "id": 14
        },
        {
          "value": 0.0,
          "id": 15
        },
        {
          "value": 0.0,
          "id": 16
        },
        {
          "value": 0.0,
          "id": 17
        },
        {
          "value": 0.0,
          "id": 18
        },
        {
          "value": 0.0,
          "id": 19
        },
        {
          "value": 0.0,
          "id": 20
        },
        {
          "value": 0.0,
          "id": 21
        },
        {
          "value": 0.0,
          "id": 22
        },
        {
          "value": 0.0,
          "id": 23
        },
        {
          "value": 0.0,
          "id": 24
        },
        {
          "value": 0.0,
          "id": 25
        },
        {
          "value": 0.0,
          "id": 26
        },
        {
          "value": 0.0,
          "id": 27
        },
        {
          "value": 0.0,
          "id": 28
        },
        {
          "value": 0.0,
          "id": 29
        },
        {
          "value": 0.0,
          "id": 30
        },
        {
          "value": 0.0,
          "id": 31
        }
      ],
      "leftModuleId": 2,
      "rightModuleId": 8370067533622772,
      "pos": [
        9,
        0
      ]
    },
```

there is no mention of any files whatsoever

**Comment by TheGuyWhoo** [comment]
I'm using latest of vcv rack v1, but the bidoo modules work correctly there. I will mention it to dev and see what he says.

Btw what's happening with the sample loading issue?

**Comment by falkTX** [comment]
> Btw what's happening with the sample loading issue?

totally different thing, and could be one of many issues. it is best discussed elsewhere

**Comment by TheGuyWhoo** [comment]
I did ask you in the appropriate place many days ago.

**Comment by falkTX** [comment]
Sure, but I dont know what is going on for that specific issue, so have nothing to add for now

**Comment by TheGuyWhoo** [comment]
Thank you, noted.

**Comment by sebastien-bouffier** [comment]
Hi folks, I confirm the issue. 
I just tested the behavior of OUAIve and Player of CF which is based on a previous version of OUAIve and OUAIve does not reload the sample when opening Reaper ... so frustrating. 
It works with VCV2 ....
Regarding liMonADe the plugin doesn't store the sample path since it stores all the wave table content in the json file.
I do not think that the issue comes from path storing.
I took a look at my Reaper test project file (rpp) and module data is stored in a non readable manner (serialized).
I anyone knows how I can log something to make a trace it will be appreciated.
I keep on searching. Stayed tuned.

**Comment by falkTX** [comment]
If this works with VCV then it can just be a Cardinal bug after all.
Do you mean it works with the standalone or the plugin version?

**Comment by sebastien-bouffier** [comment]
My modules work fine under VCV2 standalone and VST in Reaper under W10.
I installed Jack and Cardinal and samples are not loaded either in standalone mode and VST in Reaper.

**Comment by falkTX** [comment]
Turns out Cardinal is saving things fine, but I had a little thing wrong in the custom code setup for Bidoo.
Now corrected in 066344b8b0e7b107a834a3fb157e45da1b775a35

Also limonade was causing a crash if you did export wavetable and canceled, now fixed too.
